moveit_core/robot_state/src/robot_state.cpp

```diff
@@ -1981,10 +2055,58 @@ double moveit::core::RobotState::testJointSpaceJump(const JointModelGroup* group
   return percentage;
 }
 
+double moveit::core::RobotState::testJointSpaceJump(const JointModelGroup* group, std::vector<RobotStatePtr>& traj,
+                                                    double revolute_jump_threshold, double prismatic_jump_threshold)
+{
+  double percentage = 1.0;
+  bool still_valid = true;
+  const std::vector<const JointModel*>& joints = group->getActiveJointModels();
+  for (std::size_t i; i < traj.size() - 1 && still_valid; ++i)
+  {
+    for (std::size_t j = 0; j < joints.size() && still_valid; ++j)
+    {
+      const int idx = joints[j]->getFirstVariableIndex();
+      double dist = joints[j]->distance(traj[i]->position_ + idx, traj[i + 1]->position_ + idx);
+      if (joints[j]->getType() == JointModel::PRISMATIC)
+      {
+        // This is a prismatic joint
+        if (prismatic_jump_threshold > 0.0 && dist > prismatic_jump_threshold)
+        {
+          CONSOLE_BRIDGE_logDebug("Truncating Cartesian path due to detected jump in joint-space distance");
+          still_valid = false;
+        }
+      }
+      else if (joints[j]->getType() == JointModel::REVOLUTE)
+      {
+        // This is a revolute joint
+        if (revolute_jump_threshold > 0.0 && dist > revolute_jump_threshold)
+        {
+          CONSOLE_BRIDGE_logDebug("Truncating Cartesian path due to detected jump in joint-space distance", dist,
+                                  revolute_jump_threshold);
+          still_valid = false;
```

Comment:
Instead of using the `still_valid` variable, you might want directly `break` from the loop.

Reply:
I think I would need to either need to use `goto` or `return` for the double loop. I was under the impression `goto` is bad style and I thought that `still_valid` was better option than `return` since it minimized code duplication resizing the `traj` and computing the `percentage`.  I refactored the code to take still_valid out of both for loop conditionals and used the c++11 syntax for the inner loop to iterate through the joint models. For the outer loop I use the index to compute the percentage and compare each robot state in the trajectory to its predecessor so I don't think it is a good candidate for a range based loop.

Reply:
This sums up my experience, from stackoverflow: "I've never had to use a goto in C++. Ever. EVER. If there is a situation it should be used, it's incredibly rare. If you are actually considering making goto a standard part of your logic, something has flown off the tracks." https://stackoverflow.com/questions/379172/use-goto-or-not

I'm ok with still_valid vs code duplication